src/components/shared/NavCustom.js

```diff
@@ -2,29 +2,14 @@ import React, { useState, useEffect } from 'react';
 import { Nav } from 'react-bootstrap';
 import { connect } from 'react-redux';
 import PropTypes from 'prop-types';
-import { setGenre } from '../../redux/actions/criteriaActions';
-import {
-  filterByGenre,
-  setMoviesByGenre,
-  sortRating,
-  sortRelease,
-} from '../../redux/actions/moviesActions';
+import { setGenre } from '../../redux/actions/moviesActions';
 
 const NavCustom = (props) => {
-  const { dispatch, movies, moviesByCriteria, genre, sort } = props;
+  const { instGenre, genre } = props;
   const [active, setActive] = useState('All');
 
   const handleSelect = (selectedKey) => {
-    dispatch(setGenre(selectedKey));
-    dispatch(setMoviesByGenre(movies));
-
-    sort === 'vote_average'
-      ? dispatch(sortRating(moviesByCriteria))
-      : dispatch(sortRelease(moviesByCriteria));
-
-    selectedKey !== 'All'
-      ? dispatch(filterByGenre(movies.data, selectedKey))
-      : undefined;
+    instGenre(selectedKey);
```

Comment:
here is a good example of an action -  all the necessary checks or changes are done out of the component